C++/source/knapsack_problem/fractional_knapsack.cpp

```diff
@@ -0,0 +1,111 @@
+/*
+    Fractional Knapsack Problem
+    ---------------------------
+    This is a Greedy Algorithm which calculates the maximum
+    profit which can be earned from a given set of unsorted
+    objects with associated weight and profit.
+    Class implementation using Selection sort
+    for more simplicity.
+
+    Time Complexity
+    ---------------
+    O(n^2), where n is the number of objects
+
+    Space Complexity
+    ----------------
+    O(1)
+*/
+
+
+#include <iostream>
+
+using namespace std;
+
+class object {
+    public:
+        float weight;
+        float profit;
+        float p_w_ratio;
+
+        void get_data()
+        {
+            cout<<"weight:";
+            cin>>weight;
+            cout<<"profit:";
+            cin>>profit;
+        }
+        void calc_ratio ()
+        {
+            p_w_ratio = profit/weight;
+            cout<<"ratio:"<<p_w_ratio;
+            cout<<endl;
+        }
+}obj[50];
+
+void swap_data(float *x, float *y)
+{
+float temp;
+temp=*x;
```

Comment:
There's no need for this function. You can instead use the `swap` function from the `algorithm` header file.